File: src/main/java/aunmag/shooter/game/data/Sounds.java
package aunmag.shooter.game.data;

import aunmag.shooter.core.audio.Source;
import aunmag.shooter.core.utilities.UtilsAudio;
import org.jetbrains.annotations.Nullable;

public final class Sounds {

    @Nullable public static final Source soundAmbiance;
    @Nullable public static final Source soundAtmosphere;
    @Nullable public static final Source soundGameOver;

    static {
        soundAmbiance = UtilsAudio.provideSound("sounds/ambiance/birds");
        if (soundAmbiance != null) {
            soundAmbiance.setVolume(0.4f);
            soundAmbiance.setLooped(true);
        }

        soundAtmosphere = UtilsAudio.provideSound("sounds/music/gameplay_atmosphere");
        if (soundAtmosphere != null) {
            soundAtmosphere.setVolume(0.06f);
            soundAtmosphere.setLooped(true);
        }

        soundGameOver = UtilsAudio.provideSound("sounds/music/death");
        if (soundGameOver != null) {
            soundGameOver.setVolume(0.6f);
        }
    }

    private Sounds() {}

}

File: src/main/java/aunmag/shooter/game/data/Sounds.kt
package aunmag.shooter.game.data

import aunmag.shooter.core.audio.Source
import aunmag.shooter.core.utilities.UtilsAudio

val soundAmbiance = initializeSoundAmbiance()
val soundAtmosphere = initializeSoundAtmosphere()
val soundGameOver = initializeSoundGameOver()

private fun initializeSoundAmbiance(): Source? {
    val sound = UtilsAudio.provideSound("sounds/ambiance/birds")
    sound?.setVolume(0.4f)
    sound?.setLooped(true)
    return sound
}

private fun initializeSoundAtmosphere(): Source? {
    val sound = UtilsAudio.provideSound("sounds/music/gameplay_atmosphere")
    sound?.setVolume(0.06f)
    sound?.setLooped(true)
    return sound
}

private fun initializeSoundGameOver(): Source? {
    val sound = UtilsAudio.provideSound("sounds/music/death")
    sound?.setVolume(0.6f)
    return sound
}
